hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,510 @@
+    """
+
+    if finish_capture:
+        input_data.input_bit_register_64 = 1
+    else:
+        input_data.input_bit_register_64 = 0
+
+    if camera_ready:
+        input_data.input_bit_register_65 = 1
+    else:
+        input_data.input_bit_register_65 = 0
+
+    con.send(input_data)
+
+
+def _initialize_robot_sync(host: str, port: int):
+    """Set up communication with UR robot
+
+    Args:
+        host: IP address
+        port: Port number
+
+    Returns:
+        Connection to robot
+        Package containing the specific input data registers
+
+    Raises:
+        RuntimeError: If computer is not able to establish communication with robot
+    """
+
+    conf = rtde_config.ConfigFile(Path(Path.cwd() / "robot_communication_file.xml"))
+    output_names, output_types = conf.get_recipe("out")
+    input_names, input_types = conf.get_recipe("in")
+
+    con = rtde.RTDE(host, port)
+    con.connect()
+
+    # To ensure that the application is compatable with further versions of UR controller
+    if not con.negotiate_protocol_version():
+        raise RuntimeError(f"Protocol do not match")
+
+    if not con.send_output_setup(output_names, output_types, frequency=200):
+        raise RuntimeError(f"Unable to configure output")
+
+    robot_input_data = con.send_input_setup(input_names, input_types)
+
+    if not con.send_start():
+        raise RuntimeError(f"Unable to start synchronization")
+
+    print("Communication initialization completed. \n")
+
+    return con, robot_input_data
+
+
+def _save_zdf_and_pose(
+    save_dir: Path, image_num: int, frame: zivid.Frame, transform: np.array
+):
+    """Save data to folder
+
+    Args:
+        save_dir: Directory to save data
+        image_num: Image number
+        frame: Point cloud stored as .zdf
+        transform: 4x4 transformation matrix
+    """
+
+    frame.save(save_dir / f"img{image_num:02d}.zdf")
+
+    file_storage = cv2.FileStorage(
+        str(save_dir / f"pos{image_num:02d}.yaml"), cv2.FILE_STORAGE_WRITE
+    )
+    file_storage.write("PoseState", transform)
+    file_storage.release()
+
+
+def _generate_folder():
+    """Generate folder where dataset weill be stored
+
+    Returns:
+        Directory to where data will be saved
+    """
+
+    location_dir = Path(
+        Path.cwd() / "datasets" / datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    )
+    if not location_dir.is_dir():
+        location_dir.mkdir(parents=True)
+
+    return location_dir
+
+
+def _get_frame_and_transform_matrix(con: rtde, cam: zivid.Camera):
+    """Capture image with Zivid camera and read robot pose
+
+    Args:
+        con: Connection between computer and robot
+        cam: Zivid camera
+
+    Returns:
+        Zivid frame
+        4x4 tranformation matrix
+    """
+
+    frame = cam.capture()
+    robot_pose = np.array(con.receive().actual_TCP_pose)
+
+    translation = robot_pose[:3] * 1000
+    rotvec = R.from_rotvec(robot_pose[3:])
```

Comment:
I think `rotvec` should be renamed `rotation` here. The input is the rotation vector.

Perhaps something like this, for readability:
```
rotation_vector = robot_pose[3:]
rotation = Rotation.from_rotvec(rotation_vector)
```